### PR TITLE
Fetch articles by tag

### LINKED ISSUE #10: Ft: Search Articles by tags
This feature will enable plugin users get articles via tags

### PR DESCRIPTION
**Pull request by AlbertoPdRF**
Fixes #10.

This still needs work, but at least is working and not duplicating code. @OPARA-PROSPER can you take it from here?

### CODE REVIEW

File: oh-my-dev.plugin.zsh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 function fetch_api() {
   base_url="https://dev.to/api";
   url_path="/articles";
-  url=$base_url$url_path;
+  url="$base_url$url_path?tag=$1";
```
Comment (OPARA-PROSPER) [inline oh-my-dev.plugin.zsh:6]: Here

File: oh-my-dev.plugin.zsh [hunk 2]
```diff
@@ -31,7 +31,7 @@ function oh_my_dev() {
   user_exit=false;
   echo "Fetching the last 30 articles from https://dev.to";
   echo;
-  api_response=$(fetch_api);
+  api_response=$(fetch_api $1);
```
Comment (OPARA-PROSPER) [inline oh-my-dev.plugin.zsh:34]: Does oh_my_dev function require a parameter?

**Comment by AlbertoPdRF** [comment]
It's not affecting it, because I believe that if you append `?tag=` to the URL you get all the articles. But feel free to commit improvements! This is only to get started. I guess validating the tags would also be a good idea, for instance.

File: oh-my-dev.plugin.zsh (hunk 2 above)
Comment (AlbertoPdRF) [reply oh-my-dev.plugin.zsh:34]: It doesn't require it -- if you don't pass any, $1 is empty.

**Comment by OPARA-PROSPER** [comment]
True, checked it out on postman and it still works 👍

**Comment by OPARA-PROSPER** [comment]
Tried this and the error changed though...

Can you pull my PR branch and test locally?

**Comment by AlbertoPdRF** [comment]
Closing this one as #14 has been merged already.

**Pull request closed without merging**